specification/cognitiveservices/Speech.SpeechToText/main.tsp

```diff
@@ -0,0 +1,53 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi";
+import "@typespec/versioning";
+
+import "./routes/transcriptions.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using OpenAPI;
+
+@service({
+  title: "Azure Speech Services API",
+})
+@useAuth(
```

Comment:
We want to take TokenCredential derived objects at the API layer, right? This should get us that:

@useAuth(
  ApiKeyAuth<
    ApiKeyLocation.header,
    "Ocp-Apim-Subscription-Key"
  > | AadOauth2Auth<["https://cognitiveservices.azure.com/.default"]>
)